### PR TITLE
Issue/wc product model updates

### PR DESCRIPTION
**Pull request by nbradbury**
This updates the `WCProductModel` and related table with a few fields we missed the first time around:

* Downloadable files
* Download expiry
* Download limit
* External URL
* Related product IDs
* Cross-sell product IDs
* Upsell product IDs

This PR also adds the ability to parse a product's attributes, get a comma-separate list of tags and categories, and delete all products for a site.

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java [hunk 1]
```diff
@@ -638,6 +620,66 @@ private void migrateAddOn(@AddOn String addOnName, SQLiteDatabase db, int oldDbV
...
+                               + "RATING_COUNT INTEGER,"
+                               + "PARENT_ID INTEGER,"
+                               + "PURCHASE_NOTE TEXT NOT NULL,"
+                               + "CATEGORIES TEXT NOT NULL,"
+                               + "TAGS TEXT NOT NULL,"
+                               + "IMAGES TEXT NOT NULL,"
+                               + "ATTRIBUTES TEXT NOT NULL,"
+                               + "RELATED_IDS TEXT NOT NULL,"
+                               + "CROSS_SELL_IDS TEXT NOT NULL,"
+                               + "UPSELL_IDS TEXT NOT NULL,"
```
Comment (AmandaRiu) [inline fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java]: Column `UPSELL_IDS` should be `UP_SELL_IDS`, or, to better match the json response, `WCProductModel` should be changed to `upsellIds`

Comment (nbradbury) [reply fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java]: Changed in 40074db

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java [hunk 2]
```diff
@@ -583,27 +584,8 @@ private void migrateAddOn(@AddOn String addOnName, SQLiteDatabase db, int oldDbV
                     break;
                 case 56:
                     AppLog.d(T.DB, "Migrating addon " + addOnName + " to version " + (oldDbVersion + 1));
+                    // redefined in 59 below
```
[the remark this answers is not in the document]
Comment (nbradbury) [reply fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java]: Reverted in c81530e

**Comment by nbradbury** [comment]
> The db version has not been updated...The version specified in the case sequence should also be updated

Looks like this was due to a botched merge from `develop`. The version of the db was correct when I first created this PR, but then `develop` updated it again and I resolved the conflict incorrectly. Fixed in 8e380ce.

>  it would be good to add some tests for the getCommaSeparatedOptions() method

Added in 11db15a. Ready for another round!

**Pull request merged**